src/components/TagContainer.ts

```diff
@@ -0,0 +1,203 @@
+            tags: this.state.tags,
+            tagValue: this.state.tag
+        });
+    }
+
+    componentWillReceiveProps(newProps: TagContainerProps) {
+        if (newProps.mxObject !== this.props.mxObject) {
+            if (!this.props.lazyLoad) { this.fetchCurrentTags(newProps.mxObject); }
+            this.resetSubscriptions(newProps.mxObject);
+        }
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+    }
+
+    private showErrorMessage(message: string) {
+        window.mx.ui.error(message);
+    }
+
+    public static parseStyle(style = ""): {[key: string]: string} {
+        try {
+            return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
+    private getValue(attribute: string, mxObject?: mendix.lib.MxObject): string {
+        return mxObject ? (mxObject.get(attribute) as string) : "";
+    }
+
+    private isReadOnly() {
+        const { tagAttribute, editable, mxObject, readOnly } = this.props;
+        if (editable === "default" && mxObject) {
+            return readOnly || mxObject.isReadonlyAttr(tagAttribute);
+        }
+
+        return true;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.props.tagAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions() {
+        this.setState({
+            tag: this.getValue(this.props.tagAttribute, this.props.mxObject) as string
+        });
+    }
+
+    private lazyLoadTags() {
+        if (this.props.mxObject) {
+            this.fetchCurrentTags(this.props.mxObject);
+        }
+    }
+
+    private fetchCurrentTags(mxObject: mendix.lib.MxObject) {
+        const { tagAttribute, tagConstraint, tagEntity } = this.props;
+        const constraint = tagConstraint
+            ? tagConstraint.replace(/\[\%CurrentObject\%\]/gi, mxObject.getGuid())
+            : "";
+        const XPath = "//" + tagEntity + constraint;
+        mx.data.get({
```

Comment:
Wrong interpretation of the requirements
- Fetch the current selected tag over the object association
https://apidocs.mendix.com/7/client/mendix_lib_MxObject.html#getReferences

- The auto suggestion should be be constraint with an XPath.
For example `[Approve=true()]` in case you want to moderate the tags 

- When adding a tag , check the suggestion list, check system? and if available set the association between tag and context. If not available create it new and associate the context to the new create tag.

Reply:
Done